coralproject/talk
# [CORL-2818] Mod queue hotkeys (+ focus states)

## What does this PR do?
This PR addresses a bug in which the hotkeys for bam/reject/approve comment stopped working. This regression was introduced as part of an accessibility fix (CORL-2729) in which the moderation queue pulled focus immediately. This PR addresses all of these issues by adressing 2729 in a different way. Whereas before logic surrounding next/last comment was pulled from the ModerationCards up into the Queue component, this PR leaves that logic in the cards, and simply swiches `keymaster`'s "scope" to the appropriate card's `id` when toggling through ModerationCards. It initially sets the scope to the first comment's ID so that hotkeys can be registered without pulling focus from the top of the page.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## Does this PR introduce any new environment variables or feature flags?
No.

-->

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. As an admin or moderator, navigate to the moderation queue
2. Open the hotkeys help modal by pressing "Shift+/" (ie "?")
3. Observe that all hotkeys work as described (you will need some reported comments in your queue).
4. Approve and reject some comments by clicking the corresponding buttons on each card
5. Navigate to a User Drawer in the Community tab
6. Observe that comments can be approved and rejected via clicking in that view
7.With a screen reader enabled, navigate to the moderation queue route. Observe that focus remains on the top of the page until a ModerationCard is clicked on or "j" or "k" are pressed.

## Where any tests migrated to React Testing Library?
No.

## How do we deploy this PR?
No special considerations should be needed
